pkgs/development/python-modules/ruff-lsp/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, pythonOlder
+, buildPythonPackage
+, fetchPypi
+, ruff
+, pygls
+, hatchling
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "ruff-lsp";
+  version = "0.0.17";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "ruff_lsp";
+    sha256 = "sha256-5xKIhLz06y1CxLhtJpUOhcoB7HOi5mFEKINcgZs4dsU=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
+    pygls
+    typing-extensions
+  ];
+
+  postPatch = ''
+    sed -i '/"ruff>=/d' pyproject.toml
```

Comment:
Upstream mention `ruff` as [run-time](https://github.com/charliermarsh/ruff-lsp/blob/main/pyproject.toml#L38) dependency.

Reply:
ruff on pypi just installs ruff binary (written in rust) and then this library search for it on filesystem, in next sed I pass that binary as default

Reply:
ruff-lsp now searches for ruff binary in PATH, added ruff from nixpkgs to wrapper